rest/audit_test.go

```diff
@@ -46,58 +52,163 @@ func TestAuditInjectableHeader(t *testing.T) {
 		{
-			name: "diagnostic request",
-			fn:   rt.SendDiagnosticRequestWithHeaders,
+			name: "injected request data",
+			auditableAction: func(t testing.TB) {
+				headers := map[string]string{
+					requestInfoHeaderName: `{"extra":"field"}`,
+					"Authorization":       getBasicAuthHeader(requestUser, RestTesterDefaultUserPassword),
+				}
+				RequireStatus(t, rt.SendRequestWithHeaders(http.MethodGet, "/db/", "", headers), http.StatusOK)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				base.AuditIDPublicUserAuthenticated: {
+					"cid":         auditFieldValueIgnored,
+					"real_userid": map[string]any{"domain": "sgw", "user": "alice"},
+					"extra":       "field",
+				},
+			},
+		},
+		{
+			name: "rejected public request",
+			auditableAction: func(t testing.TB) {
+				RequireStatus(t, rt.SendUserRequest(http.MethodGet, "/db/", "", "incorrect"), http.StatusUnauthorized)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				base.AuditIDPublicUserAuthenticationFailed: {
+					"cid":         auditFieldValueIgnored,
+					"auth_method": "basic",
+					"db":          "db",
+					"username":    "incorrect",
+				},
+			},
+		},
+		{
+			name: "anon admin request",
+			auditableAction: func(t testing.TB) {
+				RequireStatus(t, rt.SendAdminRequest(http.MethodGet, "/db/", ""), http.StatusOK)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				// TODO: Admin auth event
+				//base.AuditIDAdminUserAuthenticated: {
+				//	"cid":         auditFieldValueIgnored,
+				//},
+				base.AuditIDReadDatabase: {
+					"cid": auditFieldValueIgnored,
+				},
+			},
 		},
-		*/
-		/* FIXME: metrics requests are not currently audited
 		{
-			name: "metrics request",
-			fn:   rt.SendMetricsRequestWithHeaders,
+			name: "authed admin request",
+			auditableAction: func(t testing.TB) {
+				if base.UnitTestUrlIsWalrus() {
+					// Skip this subtest if running with walrus - it has no support for admin auth
+					t.Skip("Skipping test that requires admin auth - Walrus not supported")
+				}
+				RequireStatus(t, rt.SendAdminRequestWithAuth(http.MethodGet, "/db/", "", base.TestClusterUsername(), base.TestClusterPassword()), http.StatusOK)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				// TODO: Admin auth event
+				//base.AuditIDAdminUserAuthenticated: {
+				//	"cid":         auditFieldValueIgnored,
+				//	"real_userid": map[string]any{"domain": "cbs", "user": base.TestClusterUsername()},
+				//},
+				base.AuditIDReadDatabase: {
+					"cid":         auditFieldValueIgnored,
+					"real_userid": map[string]any{"domain": "cbs", "user": base.TestClusterUsername()},
+				},
+			},
 		},
-		*/
 	}
 	for _, testCase := range testCases {
-		headers := map[string]string{
-			headerName:      `{"extra":"field"}`,
-			"Authorization": getBasicAuthHeader("alice", RestTesterDefaultUserPassword),
-		}
 		t.Run(testCase.name, func(t *testing.T) {
```

Comment:
I made this mistake -

```suggestion
		rt.Run(testCase.name, func(t *testing.T) {
```

It might also be that we want not pass in `t` to auditableAction and just use rt.TB()

Reply:
Will swap to `rt.Run` but will leave `t` in `auditableAction`. It's too easy to forget and rely on the parent `t` which is already in scope otherwise.